dspace-api/src/test/java/org/dspace/statistics/export/processor/ItemEventProcessorTest.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.statistics.export.processor;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import org.apache.commons.codec.CharEncoding;
+import org.dspace.AbstractDSpaceTest;
+import org.dspace.content.Item;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+/**
+ * Test class for the ItemEventProcessor
+ */
+public class ItemEventProcessorTest extends AbstractDSpaceTest {
+
+    @Mock
+    private Item item = mock(Item.class);
+
+    private ConfigurationService configurationService = DSpaceServicesFactory.getInstance().getConfigurationService();
+
+    @InjectMocks
+    ItemEventProcessor itemEventProcessor = mock(ItemEventProcessor.class, CALLS_REAL_METHODS);
+
+    private String encodedUrl;
+
+    @Before
+    public void setUp() {
+        configurationService.setProperty("stats.tracker.enabled", true);
+
+        String dspaceUrl = configurationService.getProperty("dspace.ui.url");
+        try {
+            encodedUrl = URLEncoder.encode(dspaceUrl, CharEncoding.UTF_8);
+        } catch (UnsupportedEncodingException e) {
+            throw new AssertionError("Error occurred in setup()", e);
+        }
+    }
+
+    @Test
+    /**
+     * Test the method that adds data based on the object types
+     */
+    public void testAddObectSpecificData() throws UnsupportedEncodingException {
+        itemEventProcessor.configurationService = configurationService;
```

Comment:
Again, as noted before, don't do this.  It's not a best practice to change services in other classes, even in tests.